pulumi/infra/lambda_.py

```diff
@@ -173,10 +172,10 @@ def __init__(
             opts=pulumi.ResourceOptions(parent=self),
         )
 
-        lambda_name = f"{DEPLOYMENT_NAME}-{name}"
+        name_with_deployment_prefix = f"{DEPLOYMENT_NAME}-{name}"
         self.function = aws.lambda_.Function(
             f"{name}-lambda",
-            name=override_name or name,
+            name=name_with_deployment_prefix,
```

Comment:
If we've only got the one lambda that has to have an explicit name, why don't we keep the "override name" logic (modified to fix this bug, of course), but continue to let the other lambdas get auto-generated names? That way, when we fix things so `graplctl` can just take a lambda name from Pulumi directly, the change to rip out the override name logic will only affect that one lambda, rather than all of them.

Pretty sure you could still have the log group names depend on the generated name (using a `pulumi.Output.apply`).

Reply:
Yeah, that's doable